gobblin-utility/src/main/java/gobblin/util/dataset/DatasetUtils.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.util.dataset;
+
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Predicate;
+import com.google.common.base.Strings;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Maps;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import gobblin.configuration.State;
+import gobblin.configuration.StateUtils;
+import gobblin.source.workunit.WorkUnit;
+
+
+public class DatasetUtils {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DatasetUtils.class);
+
+  public static final String DATASET = "dataset";
+
+  /**
+   * A configuration key that allows a user to specify config parameters on a dataset specific level. The value of this
+   * config should be a JSON array. Each entry should be a {@link JsonObject} and should contain a
+   * {@link com.google.gson.JsonPrimitive} that identifies the dataset. All configs in each dataset entry will
+   * be added to the {@link WorkUnit}s for that dataset.
+   *
+   * <p>
+   *   An example value could be: "[{"dataset" : "myDataset1", "writer.partition.columns" : "header.memberId"},
+   *   {"dataset" : "myDataset2", "writer.partition.columns" : "auditHeader.time"}]".
+   * </p>
+   *
+   * <p>
+   *   The "dataset" field also allows regular expressions. For example, one can specify key, value
+   *   "dataset" : "myDataset.*". In this case all datasets whose name matches the pattern "myDataset.*" will have
+   *   all the specified config properties added to their {@link WorkUnit}s. If more a dataset matches multiple
+   *   "dataset"s then the properties from all the {@link JsonObject}s will be added to their {@link WorkUnit}s.
+   * </p>
+   */
+  public static final String DATASET_SPECIFIC_PROPS = DATASET + ".specific.props";
```

Comment:
Isn't this what the `ConfigStore` is for? Typically dataset level properties should be specified in the `ConfigStore` and retrieved via the `ConfigStore` APIs.